packages/@monorepo-utils/workspaces-to-typescript-project-references/src/index.ts

```diff
@@ -54,8 +54,18 @@ export const toProjectReferences = (options: Options) => {
                 if (!absolutePathOrNull) {
                     return;
                 }
-                const tsConfigDirName = path.dirname(options.tsConfigPath ?? DEFAULT_TSCONFIGPATH);
-                const absolutePath = path.join(absolutePathOrNull, tsConfigDirName);
+                const tsConfigPath = options.tsConfigPath ?? DEFAULT_TSCONFIGPATH;
+                const tsConfigDirName = path.dirname(tsConfigPath);
+                const pathComponents = [absolutePathOrNull, tsConfigDirName];
+
+                // If the file name is not tsconfig.json (the default),
+                // then append it to the generated path
+                const tsConfigFileName = tsConfigPath.split("/").pop();
```

Comment:
```suggestion
                const tsConfigFileName = path.basename(tsConfigPath);
```
[`path.basename`](https://nodejs.org/api/path.html#path_path_basename_path_ext) is suitable.

Reply:
```
❯ node
Welcome to Node.js v14.16.1.
Type ".help" for more information.
> path.basename("/path/to/src/tsconfig-esm.json")
'tsconfig-esm.json'
```